applications/DEMApplication/tests/test_DEM_search_tolerance.py

```diff
@@ -18,123 +18,96 @@ class DEM3D_SearchToleranceMain(KratosMultiphysics.DEMApplication.DEM_analysis_s
 
     def Initialize(self):
         super().Initialize()
-        for node in self.spheres_model_part.Nodes:
-            self.initial_normal_vel = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Z)
 
-    @classmethod
     def GetMainPath(self):
         return os.path.join(os.path.dirname(os.path.realpath(__file__)), "test_search_tolerance")
 
     def GetProblemNameWithPath(self):
         return os.path.join(self.main_path, self.DEM_parameters["problem_name"].GetString())
 
+    def DoCheck1(self, y_vel, tol):
+        y_vel_ref = -5.8647604045771855
+        self.assertAlmostEqual(y_vel, y_vel_ref, delta=tol)
+
+    def DoCheck2(self, y_vel, tol):
+        y_vel_ref = -3.3860170707636836
+        self.assertAlmostEqual(y_vel, y_vel_ref, delta=tol)
+
+    def DoCheck3(self, y_vel, tol):
+        y_vel_ref = -0.5915833448135729
+        self.assertAlmostEqual(y_vel, y_vel_ref, delta=tol)
+
     def FinalizeSolutionStep(self):
         super().FinalizeSolutionStep()
-        for node in self.spheres_model_part.Nodes:
-            #reference data with freq=1 searchtolerance=0.0
-            if node.Id == 2:
-                tol = 1.0e-15
-                if np.isclose(self.time, 0.02, rtol=0.0, atol=1e-06):
-                    y_vel = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Y)
-                    print(self.time, y_vel)
-                    y_vel_ref = -5.86502139707038
-                    self.assertAlmostEqual(y_vel, y_vel_ref, delta=tol)
-
-                if np.isclose(self.time, 0.115, rtol=0.0, atol=1e-06):
-                    y_vel = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Y)
-                    print(self.time, y_vel)
-                    y_vel_ref = -3.3859516373258987
-                    self.assertAlmostEqual(y_vel, y_vel_ref, delta=tol)
-
-                if np.isclose(self.time, 0.22, rtol=0.0, atol=1e-06):
-                    y_vel = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Y)
-                    print(self.time, y_vel)
-                    y_vel_ref = -0.5929799879392164
-                    self.assertAlmostEqual(y_vel, y_vel_ref, delta=tol)
+        node = self.spheres_model_part.GetNode(2)
+        #reference data with freq=1 searchtolerance=0.0
+        tol = 1.0e-15
+
+        if np.isclose(self.time, 0.02002, rtol=0.0, atol=1e-06):
+            y_vel = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Y)
+            self.DoCheck1(y_vel, tol)
+            self.check_mark_1 = True
 
+        if np.isclose(self.time, 0.11500999999999056, rtol=0.0, atol=1e-06):
+            y_vel = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Y)
+            self.DoCheck2(y_vel, tol)
+            self.check_mark_2 = True
+
+        if np.isclose(self.time, 0.22001000000000834, rtol=0.0, atol=1e-06):
```

Comment:
I'd use `STEP` in the ProcessInfo instead.

Reply:
Yeah, maybe STEP is better here, no need to use isclose